## Fundamentals of Data Analytics assignment #1 re Anscombe quarter dataset
Mary McHale.

Background : Anscoombe quarter dataset is named after Francis Anscombe which contains 4 sets of data comprised of eleven pairs of x and y.
They have the same statistics when analysed but appear very differnet when displayed as a chart.
#Reference : https://medium.com/datadriveninvestor/anscombes-quartet-12649db7eac0

## Importing the required libraries via the Pandas package which contains fuunctions to assist in analysing data. 
Pandas is intended for "real world data analysis in Python." It is an open source data analysis/manipulation tool
available in all languages. 
Command\functions start as "pd." and it reads the data file.
Reference : https://pandas.pydata.org/


In [16]:
import pandas as pd

In [17]:
# The 4 sources of data (quartet) representing the 4 days which need to be 'read' by the Pandas package, hence.".pd". 
# They are then labeled data1 to data4.

data1 = pd.read_csv("day1.csv")
data2 = pd.read_csv("day2.csv")
data3 = pd.read_csv("day3.csv")
data4 = pd.read_csv("day4.csv")

print("This is first dataset")
print(data1)
print("This is second dataset")
print(data2)
print("This is third dataset")
print(data3)
print("This is fourth dataset")
print(data4)

This is first dataset
    x1     y1
0   10   8.04
1    8   6.95
2   13   7.58
3    9   8.81
4   11   8.33
5   14   9.96
6    6   7.24
7    4   4.26
8   12  10.84
9    7   4.82
10   5   5.68
This is second dataset
    x2    y2
0   10  9.14
1    8  8.14
2   13  8.74
3    9  8.77
4   11  9.26
5   14  8.10
6    6  6.13
7    4  3.10
8   12  9.13
9    7  7.26
10   5  4.74
This is third dataset
    x3     y3
0   10   7.46
1    8   6.77
2   13  12.74
3    9   7.11
4   11   7.81
5   14   8.84
6    6   6.08
7    4   5.39
8   12   8.15
9    7   6.42
10   5   5.73
This is fourth dataset
    x4     y4
0    8   6.58
1    8   5.76
2    8   7.71
3    8   8.84
4    8   8.47
5    8   7.04
6    8   5.25
7   19  12.50
8    8   5.56
9    8   7.91
10   8   6.89


In [5]:
# Use of "space and \n" starts a new row after each calculation to make reading the results much easier.
# The "axis=0" in the command is to show whether the data is to be analysed columnwise or rowwise. 
# For this dataset it is row wise, therefore "axis=0". Columnwise would be axis=1.
# Reference : https://pandas.pydata.org/pandas-docs/stable/index.html

print("Mean \n",data1.mean(axis=0))
print("Min \n",data1.min(axis=0))
print("Max \n",data1.max(axis=0))
print("Median \n",data1.median(axis=0))
print("Std Dev. \n",data1.std(axis=0))

Mean 
 x1    9.000000
y1    7.500909
dtype: float64
Min 
 x1    4.00
y1    4.26
dtype: float64
Max 
 x1    14.00
y1    10.84
dtype: float64
Median 
 x1    9.00
y1    7.58
dtype: float64
Std Dev. 
 x1    3.316625
y1    2.031568
dtype: float64


In [6]:
print("Mean \n",data2.mean(axis=0))
print("Min \n",data2.min(axis=0))
print("Max \n",data2.max(axis=0))
print("Median \n",data2.median(axis=0))
print("Std Dev. \n",data2.std(axis=0))

Mean 
 x2    9.000000
y2    7.500909
dtype: float64
Min 
 x2    4.0
y2    3.1
dtype: float64
Max 
 x2    14.00
y2     9.26
dtype: float64
Median 
 x2    9.00
y2    8.14
dtype: float64
Std Dev. 
 x2    3.316625
y2    2.031657
dtype: float64


In [7]:
print("Mean \n",data3.mean(axis=0))
print("Min \n",data3.min(axis=0))
print("Max \n",data3.max(axis=0))
print("Median \n",data3.median(axis=0))
print("Std Dev. \n",data3.std(axis=0))

Mean 
 x3    9.0
y3    7.5
dtype: float64
Min 
 x3    4.00
y3    5.39
dtype: float64
Max 
 x3    14.00
y3    12.74
dtype: float64
Median 
 x3    9.00
y3    7.11
dtype: float64
Std Dev. 
 x3    3.316625
y3    2.030424
dtype: float64


In [8]:
print("Mean \n",data4.mean(axis=0))
print("Min \n",data4.min(axis=0))
print("Max \n",data4.max(axis=0))
print("Median \n",data4.median(axis=0))
print("Std Dev. \n",data4.std(axis=0))

Mean 
 x4    9.000000
y4    7.500909
dtype: float64
Min 
 x4    8.00
y4    5.25
dtype: float64
Max 
 x4    19.0
y4    12.5
dtype: float64
Median 
 x4    8.00
y4    7.04
dtype: float64
Std Dev. 
 x4    3.316625
y4    2.030579
dtype: float64


## Data analysis
Having analysed the data, it is however visually hard to compare the results on screen when there are four data sets listed. 
After searching for a more compact analysis Panda function - ".describe()" allows the dataset to be summarised concisely and contains the min, max, mean, standard deviation and other analytical results of the data
Reference : https://pandas.pydata.org/pandas-docs/stable/index.html


In [9]:
summary1=data1.describe()
summary2=data2.describe()
summary3=data3.describe()
summary4=data4.describe()

print(summary1,"\n",summary2,"\n", summary3,"\n", summary4)

# Use of the "List" function allows the data to be shown with the analysis of all 4 datasets in a list format shown one after the other.

list_of_summaries = [summary1,summary2,summary3,summary4]

# Use of "concat" function to display all the 4 datasets together
result=pd.concat(list_of_summaries)

              x1         y1
count  11.000000  11.000000
mean    9.000000   7.500909
std     3.316625   2.031568
min     4.000000   4.260000
25%     6.500000   6.315000
50%     9.000000   7.580000
75%    11.500000   8.570000
max    14.000000  10.840000 
               x2         y2
count  11.000000  11.000000
mean    9.000000   7.500909
std     3.316625   2.031657
min     4.000000   3.100000
25%     6.500000   6.695000
50%     9.000000   8.140000
75%    11.500000   8.950000
max    14.000000   9.260000 
               x3         y3
count  11.000000  11.000000
mean    9.000000   7.500000
std     3.316625   2.030424
min     4.000000   5.390000
25%     6.500000   6.250000
50%     9.000000   7.110000
75%    11.500000   7.980000
max    14.000000  12.740000 
               x4         y4
count  11.000000  11.000000
mean    9.000000   7.500909
std     3.316625   2.030579
min     8.000000   5.250000
25%     8.000000   6.170000
50%     8.000000   7.040000
75%     8.000000   8.190000
max    19.0000

In [20]:
## The results
result


x1         x2         x3         x4         y1         y2  \
count  11.000000        NaN        NaN        NaN  11.000000        NaN   
mean    9.000000        NaN        NaN        NaN   7.500909        NaN   
std     3.316625        NaN        NaN        NaN   2.031568        NaN   
min     4.000000        NaN        NaN        NaN   4.260000        NaN   
25%     6.500000        NaN        NaN        NaN   6.315000        NaN   
50%     9.000000        NaN        NaN        NaN   7.580000        NaN   
75%    11.500000        NaN        NaN        NaN   8.570000        NaN   
max    14.000000        NaN        NaN        NaN  10.840000        NaN   
count        NaN  11.000000        NaN        NaN        NaN  11.000000   
mean         NaN   9.000000        NaN        NaN        NaN   7.500909   
std          NaN   3.316625        NaN        NaN        NaN   2.031657   
min          NaN   4.000000        NaN        NaN        NaN   3.100000   
25%          NaN   6.500000        NaN        NaN        NaN   6.695000   
50%          NaN   9.000000        NaN        NaN        NaN   8.140000   
75%          NaN  11.500000        NaN        NaN        NaN   8.950000   
max          NaN  14.000000        NaN        NaN        NaN   9.260000   
count        NaN        NaN  11.000000        NaN        NaN        NaN   
mean         NaN        NaN   9.000000        NaN        NaN        NaN   
std          NaN        NaN   3.316625        NaN        NaN        NaN   
min          NaN        NaN   4.000000        NaN        NaN        NaN   
25%          NaN        NaN   6.500000        NaN        NaN        NaN   
50%          NaN        NaN   9.000000        NaN        NaN        NaN   
75%          NaN        NaN  11.500000        NaN        NaN        NaN   
max          NaN        NaN  14.000000        NaN        NaN        NaN   
count        NaN        NaN        NaN  11.000000        NaN        NaN   
mean         NaN        NaN        NaN   9.000000        NaN        NaN   
std          NaN        NaN        NaN   3.316625        NaN        NaN   
min          NaN        NaN        NaN   8.000000        NaN        NaN   
25%          NaN        NaN        NaN   8.000000        NaN        NaN   
50%          NaN        NaN        NaN   8.000000        NaN        NaN   
75%          NaN        NaN        NaN   8.000000        NaN        NaN   
max          NaN        NaN        NaN  19.000000        NaN        NaN   

              y3         y4  
count        NaN        NaN  
mean         NaN        NaN  
std          NaN        NaN  
min          NaN        NaN  
25%          NaN        NaN  
50%          NaN        NaN  
75%          NaN        NaN  
max          NaN        NaN  
count        NaN        NaN  
mean         NaN        NaN  
std          NaN        NaN  
min          NaN        NaN  
25%          NaN        NaN  
50%          NaN        NaN  
75%          NaN        NaN  
max          NaN        NaN  
count  11.000000        NaN  
mean    7.500000        NaN  
std     2.030424        NaN  
min     5.390000        NaN  
25%     6.250000        NaN  
50%     7.110000        NaN  
75%     7.980000        NaN  
max    12.740000        NaN  
count        NaN  11.000000  
mean         NaN   7.500909  
std          NaN   2.030579  
min          NaN   5.250000  
25%          NaN   6.170000  
50%          NaN   7.040000  
75%          NaN   8.190000  
max          NaN  12.500000

## The result of the analysis:
That the statistical values for the mean, minimum and maximum are the same for most of the 'x' datasets, for example the max is 14 for x1, x2 and x3. Mean of x is 9 and mean of y is 7.50 for all 4 datasets. 

## Plotting the data:
Next step requires plotting the data to see how it looks when plotted as a graph.

## Importing Matplotlib as it allows data to be charted, graphed,plotted easily with the minimum of commands. 
John D. Hunter was the original author. 
Matplotlib is a library in Jupyter Notebooks which plots the data.
First, need to use the command "import matplotlib.pyplot" into Jupyter Notebook
Reference : https://matplotlib.org/

In [11]:
import matplotlib.pyplot as plt

## Matplotlib then plots the data based on the intersection of points x and y. 
Command "..'ro')" attributes colour to the plot, for example ro = red for data1, o = blue for data2 which is the default colour,
'go' = green for data3 and 'yo' = yellow for data4 chart below.
The command\function is plt.plot for the dataset. PLOT sorts the data automatically. 
The double set of square brackets are used to access the column of data.
Function plt.show() prints the output to the screen.

In [ ]:
lt.plot(data1[['x1']],data1[['y1']], 'ro')
plt.show()

DATA1 chart displays the data following a linear pattern.

In [ ]:
plt.plot(data2[['x2']],data2[['y2']], 'o')
plt.show()

DATA2 chart displays the data as an upward curve reaching a peak and then declining with no outliers.

In [ ]:
plt.plot(data3[['x3']],data3[['y3']], 'go')
plt.show()

DATA3 chart displays as a linear line trending upwards with one outlier or datapoint which does not follow the trend. 

In [ ]:
plt.plot(data4[['x4']],data4[['y4']], 'yo')
plt.show()

DATA4 chart displays the data in a vertical line with one outlier.

## Summary:
The Anscombe quartet dataset shows the importance of using both statistics and plotting of data visually for data analysis to ensure comprehensice understanding of the data.
